k8s/src/main/scala/io/buoyant/k8s/v1.scala

```diff
@@ -15,43 +15,74 @@ package object v1 {
 
   @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
   @JsonSubTypes(Array(
-    new JsonSubTypes.Type(value = classOf[Added], name = "ADDED"),
-    new JsonSubTypes.Type(value = classOf[Modified], name = "MODIFIED"),
-    new JsonSubTypes.Type(value = classOf[Deleted], name = "DELETED"),
-    new JsonSubTypes.Type(value = classOf[Error], name = "ERROR")
+    new JsonSubTypes.Type(value = classOf[EndpointsAdded], name = "ADDED"),
+    new JsonSubTypes.Type(value = classOf[EndpointsModified], name = "MODIFIED"),
+    new JsonSubTypes.Type(value = classOf[EndpointsDeleted], name = "DELETED"),
+    new JsonSubTypes.Type(value = classOf[EndpointsError], name = "ERROR")
   ))
   sealed trait EndpointsWatch extends Watch[Endpoints]
-  case class Added(
+  case class EndpointsAdded(
     `object`: Endpoints
   ) extends EndpointsWatch with Watch.Added[Endpoints]
 
-  case class Modified(
+  case class EndpointsModified(
     `object`: Endpoints
   ) extends EndpointsWatch with Watch.Modified[Endpoints]
 
-  case class Deleted(
+  case class EndpointsDeleted(
     `object`: Endpoints
   ) extends EndpointsWatch with Watch.Deleted[Endpoints]
 
-  case class Error(
+  case class EndpointsError(
     @JsonProperty(value = "object") status: Status
   ) extends EndpointsWatch with Watch.Error[Endpoints]
 
   implicit object EndpointsDescriptor extends ObjectDescriptor[Endpoints, EndpointsWatch] {
     def listName = "endpoints"
-    def toWatch(e: Endpoints) = Modified(e)
+    def toWatch(e: Endpoints) = EndpointsModified(e)
+  }
+
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+  @JsonSubTypes(Array(
+    new JsonSubTypes.Type(value = classOf[ServiceAdded], name = "ADDED"),
+    new JsonSubTypes.Type(value = classOf[ServiceModified], name = "MODIFIED"),
+    new JsonSubTypes.Type(value = classOf[ServiceDeleted], name = "DELETED"),
+    new JsonSubTypes.Type(value = classOf[ServiceError], name = "ERROR")
+  ))
+  sealed trait ServiceWatch extends Watch[Service]
```

Comment:
tioli: i know this naming comes from the prior api, but might it be more appropriate to s/Watch/Event/ ?